lib/buffer.js

```diff
@@ -624,10 +624,30 @@ function stringSlice(buf, encoding, start, end) {
   throw new ERR_UNKNOWN_ENCODING(encoding);
 }
 
-Buffer.prototype.copy =
-  function copy(target, targetStart, sourceStart, sourceEnd) {
-    return _copy(this, target, targetStart, sourceStart, sourceEnd);
-  };
+function validateOffsetType(val, name) {
+  if (val) {
+    if (typeof val !== 'number') {
+      throw new ERR_INVALID_ARG_TYPE(name, ['Number'], val);
+    }
+    if (val < 0) {
+      throw new ERR_OUT_OF_RANGE(name, '>= 0', val);
+    }
+  } else {
+    val = 0;
+  }
+  return val;
+}
+
+Buffer.prototype.copy = function(target, targetStart, sourceStart, sourceEnd) {
+  if (!isUint8Array(target)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'otherBuffer', ['Buffer', 'Uint8Array'], target);
```

Comment:
I believe it should be `target` to match documentation and stack traces, and not `otherBuffer`.

Reply:
ack